src/analyze/analyze.c

```diff
@@ -318,13 +318,22 @@ static int acquire_boot_times(sd_bus *bus, struct boot_times **bt) {
 }
 
 static void free_host_info(struct host_info *hi) {
-        free(hi->hostname);
-        free(hi->kernel_name);
-        free(hi->kernel_release);
-        free(hi->kernel_version);
-        free(hi->os_pretty_name);
-        free(hi->virtualization);
-        free(hi->architecture);
+        if (hi == NULL)
+                return;
+        if (hi->hostname != NULL)
+                free(hi->hostname);
+        if (hi->kernel_name != NULL)
+                free(hi->kernel_name);
+        if (hi->kernel_release != NULL)
+                free(hi->kernel_release);
+        if (hi->kernel_version != NULL)
+                free(hi->kernel_version);
+        if (hi->os_pretty_name != NULL)
+                free(hi->os_pretty_name);
+        if (hi->virtualization != NULL)
+                free(hi->virtualization);
+        if (hi->architecture != NULL)
+                free(hi->architecture);
```

Comment:
It wasn't needed so far, so I don't see the point. However, I would agree on a patch that uses `DEFINE_TRIVIAL_CLEANUP_FUNC` and `_cleanup_` annotations in all the functions that use `struct host_info`, so we can also get rid of the stupid `goto` logic. Just add that as another patch to the reworked PR if you wish.